app/controllers/notifications_controller.rb

```diff
@@ -0,0 +1,90 @@
+class NotificationsController < ApplicationController
+  before_action :show_notification_count, only: :index
+
+  def index
+    if current_user.user_type == "taskee" then taskee_notifications end
+    if current_user.user_type == "tasker" then tasker_notifications end
+  end
+
+  def show
+    request = TaskManagement.find params[:id]
+    request.update_attribute(:viewed, true)
+    title = params[:title]
+    description = request.task_desc
+    date = pad_date(request.start_time.strftime("%B %e")) << " of this year"
+    time = format_for_view(request.start_time, request.end_time)
+    amount = request.amount
+
+    render json: {
+      title: title,
+      description: description,
+      date: date,
+      time: time,
+      amount: amount
+    }
+  end
+
+  def update
+    record = TaskManagement.find params[:id]
+    if record.update_attribute(:status, params[:status])
+      render json: { message: "success" }
+    end
+  end
+
+  private
+
+  def notifications_for_taskee
+    @notifications ||= TaskManagement.where(taskee_id: current_user.id,
+                                            status: "inactive").
+                       order(viewed: :asc, created_at: :desc).
+                       select(:id, :task_id, :tasker_id, :viewed)
+  end
+
+  def notifications_for_tasker
+  end
+
+  def taskee_notifications
+    all_notifications_seen_for(current_user)
+    notifications_for_taskee
+  end
+
+  def tasker_notifications
+    all_notifications_seen_for(current_user)
+    notifications_for_tasker
+  end
+
+  def all_notifications_seen_for(user)
+    query = user.user_type == "tasker" ? "tasker_id" : "taskee_id"
+    attribute =
+      user.user_type == "tasker" ? "tasker_notified" : "taskee_notified"
+    TaskManagement.where(query => user.id).each do |obj|
```

Comment:
Take a look at `update_all` method for batch upload. Also this query belongs to the model `TaskManagement`.